bolt/bolt_router.py

```diff
@@ -220,19 +224,22 @@ def send(self, *args, **kwargs):
         # content, it's just the SHA of an empty string and it's always the value below.
         # https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
         request.headers['X-Amz-Content-Sha256'] = 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
-
-
         self._auth.add_auth(request)
 
         for key in ["X-Amz-Date", "Authorization", "X-Amz-Security-Token", "X-Amz-Content-Sha256"]:
           if request.headers.get(key):
             prepared_request.headers[key] = request.headers[key]
         prepared_request.headers['X-Bolt-Auth-Prefix'] = self._prefix
 
-        # send this request with our custom session options
-        # if an AWSResponse is returned directly from a `before-send` event handler function, 
-        # botocore will use that as the response without making its own request.
-        return BoltSession(self._hostname).send(prepared_request)
+        try: 
+          bolt_response =  BoltSession(self._hostname, verify=ssl_verify).send(prepared_request)
+          if 400 <= bolt_response.status_code < 500:
+              logger.info"bolt request failed - 4xx - falling back to aws", extra={"status_code": bolt_response.status_code})
+              return URLLib3Session().send(incoming_request)
+          return bolt_response
+        except Exception as e:
```

Comment:
Why do we need the exception clause?

Reply:
because something when construction the session can fail, like the ssl things based on the setting of `verify` argument -- its similar to how we failover on 4xx to s3 in sidekick but also if a panic happens.